lib/common/ipc_controld.c

```diff
@@ -126,7 +126,7 @@ set_node_info_data(pcmk_controld_api_reply_t *data, xmlNode *msg_data)
     data->data.node_info.have_quorum = pcmk__xe_attr_is_true(msg_data, XML_ATTR_HAVE_QUORUM);
     data->data.node_info.is_remote = pcmk__xe_attr_is_true(msg_data, XML_NODE_IS_REMOTE);
     crm_element_value_int(msg_data, XML_ATTR_ID, &(data->data.node_info.id));
-    data->data.node_info.uuid = crm_element_value(msg_data, XML_ATTR_UUID);
+    data->data.node_info.uuid = crm_element_value(msg_data, XML_ATTR_ID);
```

Comment:
Again how you have it here is good (node_info.uuid as id string, and node_info.id as integer equivalent). Theoretically we could limit node_info.id to Corosync nodes but it doesn't matter currently. I wouldn't bother with it (except maybe a comment) until we can refactor crm_peer_t to isolate layer-specific data better.

Reply:
Sounds good, I'll add a comment. I tested with `crm_node --cluster-id` on a remote node, and it prints `-1`, so that's reasonable to indicate invalid.

Also interesting is that it doesn't include itself in the list of all known members, since it's not a Corosync member. Maybe desirable in and of itself but an clarification in the help output might be warranted. Anyway, out of scope.

Reply:
Also thinking out loud, we might want to add a `crm_node` option to get the `uuid` string field as a string. Currently we have options for integer `id` and string `uname`.

Reply:
> Sounds good, I'll add a comment. I tested with crm_node --cluster-id on a remote node, and it prints -1, so that's reasonable to indicate invalid.

You could make a case for printing the node name or nothing in that case, but unless a user is interested I wouldn't bother.

> Also interesting is that it doesn't include itself in the list of all known members, since it's not a Corosync member. Maybe desirable in and of itself but an clarification in the help output might be warranted. Anyway, out of scope.

That's correct, -l lists cluster members only. The help should make that clearer.

> Also thinking out loud, we might want to add a crm_node option to get the uuid string field as a string. Currently we have options for integer id and string uname.

Not necessary, that's an implementation detail rather than an interface issue. The -i/--cluster-id option should display the cluster ID as appropriate to the cluster layer, and for Corosync an integer is fine. UUID per se isn't something we need to expose to the user. I suspect it will go away in the next refactor.